src/map/utils/battleutils.cpp

```diff
@@ -5316,14 +5316,17 @@ namespace battleutils
     // Calculate TP generated by spell for Occult Acumen trait
     int16 CalculateSpellTP(CBattleEntity* PEntity, CSpell* PSpell)
     {
-        if (PSpell->getSkillType() == SKILLTYPE::SKILL_ELE || PSpell->getSkillType() == SKILLTYPE::SKILL_DRK)
+        // Players onry
+        if (PEntity->objtype == TYPE_PC)
         {
-            if (CCharEntity* PChar = dynamic_cast<CCharEntity*>(PEntity))
+            if (PSpell->getSkillType() == SKILLTYPE::SKILL_ELE || PSpell->getSkillType() == SKILLTYPE::SKILL_DRK)
             {
+                CCharEntity* PChar = dynamic_cast<CCharEntity*>(PEntity);
```

Comment:
almost!
we want `static_cast` here instead of `dynamic_cast` since we already know from `PEntity->objtype == TYPE_PC` this is a `CCharEntity`, and we don't need `dynamic_cast` to check that again